Upgrade build configuration

dokmic opened the pull request:
This pull-request fixes the following:
- Add ECMAScript Modules support.
- Upgrade dependencies.
- Migrate to the automated builds.
- Migrate to eslint.
- Migrate to prettier.
- Migrate to rollup.
- Migrate to the Airbnb style guide.

Closes #11 .

chuwy commented:
Hey @dokmic, sorry for the delay - I'll review it today!

dokmic commented:
Hey @chuwy, did you have a chance to take a look at the change?

paulboocock commented on src/event.ts:
```diff
@@ -67,13 +68,13 @@ export interface Event {
   mkt_content: string;
   mkt_campaign: string;
-  contexts: any;
+  contexts: unknown;
```
Switching to unknown is a nice idea, but this is a backwards incompatible change. I feel this change pushes this beyond a PATCH release and is at least a MINOR, I appreciate we are still on MAJOR v0 so I think this is fine getting a MINOR bump to 0.2.0.

paulboocock commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "snowplow-analytics-sdk",
-  "version": "0.1.0",
+  "version": "0.1.1",
```
0.2.0? Feels like a big change for a patch (plus changes to the Event interface).

goodits replied:
Is this a symptom that it should be a 1.x.x repo already?

paulboocock commented on src/index.ts:
```diff
@@ -10,60 +10,57 @@
+  const errors = [] as string[];
+  const output = Object.keys(EVENT_STRUCTURE)
+    .reduce(
+      (output, key, index) => {
```
This line fails linting locally for me due to the new line after `.reduce(` and the new line at the bottom of the function before `[] as Field[],`. Running --fix reshuffles this section a little and fixes linting locally.
I'm not entirely sure why this passes on Travis and fails locally however, I have tested the altered version on Travis and it also passes: https://travis-ci.org/snowplow-incubator/snowplow-js-analytics-sdk/builds/617177080

dokmic commented:
Hey @chuwy and @paulboocock,

Thank you for the review. I've fixed what you discovered:
- Updated the `CHANGELOG` to be upside down;
- Did a minor bump (`0.2.0` now);
- Fixed formatting causing failed linting.
